### PR TITLE
updated CNN and DNN unit tests

### PR DESCRIPTION
**Pull request by kamuda1**
This PR updates the CNN and DNN unit tests for the Tensorflow classes in the `model_classes.py` module. Tests use fixtures to create models with dense connections and convolutional filters of all ones. Training tests and tests for the CAE and DAE to come.

Tests include checking if:
- dropout changes network output when applied (during training)
- dropout changes network output when not applied (during inference)
- l2 regularization adds to the cost function
- output values to unit inputs (inputs of all ones) make sense

### CODE REVIEW

File: annsa/tests/test_training_dnn.py [hunk 1]
```diff
@@ -1,80 +1,89 @@
...
-        verbose=1,
-        print_errors=0,
-        obj_cost=model.cross_entropy,
-        earlystop_cost_fn=model.f1_error,
-        data_augmentation=model.default_data_augmentation,)
-    pass
+    # forward pass to initialize dnn weights
+    model.forward_pass(np.ones([1, 1024]), training=False)
+    # set weights to ones
+    weight_ones = []
```
Comment (samgdotson) [inline annsa/tests/test_training_dnn.py]:
```suggestion
    weight_ones = [np.ones(weight.shape) if (index % 2 == 0) else weight for index,
    weight in enumerate(model.get_weights())]
```
Same as above

**Review by samgdotson — APPROVED** [review]
These tests look good. My comments were either a clarification or just my opinion.

File: annsa/model_classes.py [hunk 2]
```diff
@@ -608,7 +608,6 @@ def forward_pass(self, input_data, training):
 
         """
         x = self.scaler.transform(input_data)
-        x = tf.reshape(x, [-1, 1, x.shape[1]])
```
[the remark this answers is not in the document]
Comment (kamuda1) [reply annsa/model_classes.py]: I'll include a test in this PR to ensure this doesn't affect training and inference. Rerunning things in my notebooks with this worked fine. The shape of this only affects the shape of the output from `forward_pass`, which should be fine if the loss functions do their own reshaping to match the targets.

File: annsa/tests/test_training_cnn1d.py [hunk 3]
```diff
@@ -1,92 +1,128 @@
...
-        verbose=1,
-        print_errors=0,
-        obj_cost=model.cross_entropy,
-        earlystop_cost_fn=model.f1_error,
-        data_augmentation=model.default_data_augmentation,)
-    pass
+    # forward pass to initialize cnn1d weights
+    model.forward_pass(np.ones([1, 1024]), training=False)
+    # set weights to ones
+    weight_ones = []
```
[the remark this answers is not in the document]
Comment (kamuda1) [reply annsa/tests/test_training_cnn1d.py]: Pretty! I want to incorporate more clear list comprehension in my code, so I'd happily include this.

File: annsa/model_classes.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kamuda1) [reply annsa/model_classes.py]: Training tests added, looks like the reshape wasn't necessary (hurray unit testing for helping us find this!).

File: annsa/tests/test_training_cnn1d.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kamuda1) [reply annsa/tests/test_training_cnn1d.py]: Added! Much cleaner. I'll look out for more opportunities to use list comp, thanks Sam!

**Comment by samgdotson** [comment]
Everything looks good, merging PR
